CheckSysVinitOnSystemd.py

```diff
@@ -0,0 +1,73 @@
+#############################################################################
+# File          : CheckSysVinitOnSystemd.py
+# Package       : rpmlint
+# Author        : Werner Fink
+# Created on    : Tue Feb 21 17:34:50 2017
+# Purpose       : Check on systemd systems for required insserv package
+#############################################################################
+
+from Filter import addDetails, printError, printWarning
+import AbstractCheck
+import Config
+import string
+import os
+import stat
+import Pkg
+
+insserv_tag = 'insserv-is-required-on-systemd-based'
+etcinit_tag = 'etc-init-found-on-systemd-based'
+bootscr_tag = 'sysv-boot-script-found-on-systemd-based'
+
+#
+# Should be set in the global configuration file
+#
+#Config.setBadness(insserv_tag, 10000)
+#Config.setBadness(etcinit_tag, 10000)
+#Config.setBadness(bootscr_tag, 10000)
+
+if Config.info:
+    addDetails(
+insserv_tag,
+'''packages shall not require insserv on a system based on systemd
+therefor please consider to remove such dependencies''',
+etcinit_tag,
+'''packages shall not include /etc/init.d/ on a system based on systemd
+as this is deprecated, please remove''',
+bootscr_tag,
+'''packages shall not install SysVinit boot scripts on a system based on systemd
+therefor please convert all boot scripts into systemd unit files''',
+)
+
+class CheckSysVinitOnSystemd(AbstractCheck.AbstractCheck):
+    def __init__(self):
+        self.map = []
+        AbstractCheck.AbstractCheck.__init__(self, 'CheckSysVinitOnSystemd')
+
+    def check(self, pkg):
+        if pkg.isSource():
+            return
+
+        for req in pkg.requires():
+            if req[0].startswith('insserv'):
```

Comment:
shouldn't that be a comparison rather than a "startswith"? a requires on "insserv_compat" should be totally fine.